test/unit/BPool/BPool_JoinPool.t.sol

```diff
@@ -8,30 +8,23 @@ import {BNum} from 'contracts/BNum.sol';
 import {IBPool} from 'interfaces/IBPool.sol';
 
 contract BPoolJoinPool is BPoolBase {
-  uint256 public poolAmountOut = 10e18;
-  // Assets Under Management
-  uint256 public tokenAUM = 10e18;
-  uint256 public secondTokenAUM = 30e18;
+  // Valid scenario
+  uint256 public SHARE_PROPORTION = 10;
+  uint256 public poolAmountOut = INIT_POOL_SUPPLY / SHARE_PROPORTION;
+  uint256 public token0Balance = 10e18;
+  uint256 public token1Balance = 30e18;
+  uint256[] maxAmountsIn;
 
   // when minting n pool shares, enough amount X of every token t should be provided to statisfy
   // Xt = n/BPT.totalSupply() * t.balanceOf(BPT)
-  // in this scenario n = 10, totalSupply = 100
-
-  // for t = token -> t.balanceOf(BPT) = tokenAUM = 10
-  // therefore Xt = 10/100*10 = 10
-  uint256 public requiredTokenIn = 1e18;
-  // for t = secondToken -> t.balanceOf(BPT) = secondTokenAUM = 30
-  // therefore Xt = 10/100*30 = 30
-  uint256 public requiredSecondTokenIn = 3e18;
-  // put another way, if the current 100 shares represent 30 secondToken, then
-  // to mint another 10, user should provide 3 secondToken so ratio stays the same.
+  uint256 public requiredTokenIn = token0Balance / SHARE_PROPORTION;
+  uint256 public requiredSecondTokenIn = token1Balance / SHARE_PROPORTION;
 
   function setUp() public virtual override {
     super.setUp();
     bPool.set__finalized(true);
-    // simulate expected finalize outcome
-    bPool.call__mintPoolShare(INIT_POOL_SUPPLY);
-    bPool.call__pushPoolShare(deployer, INIT_POOL_SUPPLY);
+    // simulate an amount of pool shares (minted to the pool for easy burning)
+    deal(address(bPool), address(bPool), INIT_POOL_SUPPLY, true);
```

Comment:
I do not like creating the shares in one way but then burning them in another, I'd call `call__mintPoolShare` which should do the exact same thing, and is also consistent with then calling `call__burnPoolShare` to remove them

Reply:
hmmmm ok, can agree, but let's not push them to the deployer, since that has no effect at all, despite being the normal initial scenario